app/src/main/java/com/google/android/sambadocumentsprovider/provider/SambaDocumentsProvider.java

```diff
@@ -131,20 +136,22 @@ public void onMountedServerChange() {
   };
 
   private ShareManager mShareManager;
-  private SmbClient mClient;
+  private SmbFacade mClient;
   private ByteBufferPool mBufferPool;
   private DocumentCache mCache;
   private TaskManager mTaskManager;
+  private StorageManager mStorageManager;
 
   @Override
   public boolean onCreate() {
     final Context context = getContext();
-    mClient = SambaProviderApplication.getSambaClient(context);
+    mClient = (SmbFacade) SambaProviderApplication.getSambaClient(context);
```

Comment:
Change SambaProviderApplication.getSambaClient() to return SmbFacade to get rid of this cast. Change SambaMessageLooper as well.